mpegts-segmenter/src/analyzer.rs

```diff
@@ -176,6 +193,56 @@ impl Stream {
                                 0 => 0.0,
                                 num_units_in_tick @ _ => (sps.vui_parameters.time_scale.0 as f64 / (2.0 * num_units_in_tick as f64) * 100.0).round() / 100.0,
                             };
+                            *last_vui_parameters = Some(sps.vui_parameters);
+                        }
+                        h264::NAL_UNIT_TYPE_SUPPLEMENTAL_ENHANCEMENT_INFORMATION => {
+                            let bs = h264::Bitstream::new(nalu.iter().copied());
+                            let mut nalu = h264::NALUnit::decode(bs)?;
+
+                            if let Some(vui_params) = &last_vui_parameters {
+                                let mut rbsp = h264::Bitstream::new(&mut nalu.rbsp_byte);
+                                let sei = h264::SEI::decode(&mut rbsp)?;
+
+                                let acc: Vec<Timecode> = last_timecode.iter().cloned().collect();
+                                let timecodes = sei
+                                    .sei_message
+                                    .iter()
+                                    .flat_map(|sei_message| {
+                                        let mut bs = h264::Bitstream::new(sei_message.payload.iter().copied());
+                                        h264::PicTiming::decode(&mut bs, &vui_params).unwrap().timecodes
+                                    })
+                                    .fold(acc, |mut acc, t| {
+                                        let mut timecode = Timecode {
+                                            hours: t.hours.0,
+                                            minutes: t.minutes.0,
+                                            seconds: t.seconds.0,
+                                            frames: t.n_frames.0,
+                                        };
+                                        if let Some(previous_timecode) = acc.iter().last() {
```

Comment:
Isn't `acc` a `Vec`? Can't you just use `.last()` without iterating again?

```suggestion
                                        if let Some(previous_timecode) = acc.last() {
```